js/controller/BoardController.js

```diff
@@ -186,16 +208,48 @@ app.controller('BoardController', function ($rootScope, $scope, $stateParams, St
 		});
 	};
 
+	$scope.stackDelete = function (stack) {
+		$scope.stackservice.delete(stack.id).then(function() {
+			$scope.deletedStacks[stack.id] = stack;
+		});
+	}
+
+	$scope.stackUndoDelete = function (deletedStack) {
+		return StackService.undoDelete(deletedStack).then(function() {
+			delete $scope.deletedStacks[deletedStack.id];
+		});
+	}
+
 	$scope.cardDelete = function (card) {
-		OC.dialogs.confirm(t('deck', 'Are you sure you want to delete this card with all of its data?'), t('deck', 'Delete'), function(state) {
-			if (!state) {
-				return;
+		CardService.delete(card.id).then(function () {
+			StackService.removeCard(card);
+			$scope.deletedCards[card.id] = card;
+		});
+	};
+
+	$scope.cardUndoDelete = function (deletedCard) {
+		CardService.undoDelete(deletedCard).then(function() {
+			delete $scope.deletedCards[deletedCard.id];
+
+			var associatedDeletedStack = $scope.deletedStacks[deletedCard.stackId];
+			if(associatedDeletedStack !== undefined) {
+				OC.dialogs.confirm(
+					t('deck', 'The associated stack is deleted as well, do you want to restore it as well?'),
+					t('deck', 'Yes'),
+					function(state) {
+
+					if (state) {
```

Comment:
There is no need to handle this. If the stack gets deleted, all containing cards will also be deleted.

Reply:
Hey,

The question is about the restoration of deleted cards.

Consider a card is deleted and the stack it is associated with is deleted afterwards.

Now the question deletion is undone by the user. 

What will happen to the deleted stack?

With the current implementation, the user is asked, if the associated deleted stack should also be restored.

But if the user decicdes not to do this, the card is not shown on the board due to the lack of an associated stack.

Reply:
Hey,

I will now just not make it possible to undelete a card, in case the user decides against restoring the associated stack. This will be disclosed in the shown dialogue.

It can still be adapted, just want to finish this PR.

Reply:
Yes, let's keep it simple and don't allow restore of cards on stacks which got deleted. We can still extend this later, if it is really needed.